Components/Metrics/PointToSurfaceDistance/itkPointToSurfaceDistanceMetric.h

```diff
@@ -0,0 +1,130 @@
+    using PointIterator = typename Superclass::PointIterator;
+    using PointDataIterator = typename Superclass::PointDataIterator;
+    using InputPointType = typename Superclass::InputPointType;
+    using OutputPointType = typename Superclass::OutputPointType;
+    using CoordRepType = typename OutputPointType::CoordRepType;
+    using VnlVectorType = vnl_vector< CoordRepType >;
+    using NonZeroJacobianIndicesType = typename Superclass::NonZeroJacobianIndicesType;
+
+	   /** Constants for the pointset dimensions. */
+    itkStaticConstMacro( FixedPointSetDimension, unsigned int, TFixedPointSet::PointDimension );
+
+    using ImageType = itk::Image< float, itkGetStaticConstMacro( FixedPointSetDimension )>;
+    using InputImageType = itk::Image< unsigned char,  itkGetStaticConstMacro( FixedPointSetDimension ) >;
+    using RescalerType = itk::RescaleIntensityImageFilter< ImageType, ImageType >;
+    using DtFilterType = itk::DanielssonDistanceMapImageFilter< InputImageType, ImageType, ImageType >;
+    using NegateFilterType = itk::InvertIntensityImageFilter< InputImageType, InputImageType >;
+    using AddImageFilt = itk::AddImageFilter< ImageType, ImageType, ImageType >;
+    using SegReaderType =itk::ImageFileReader< InputImageType >;
+    using DTReaderType = itk::ImageFileReader< ImageType  >;
+    using WriterType = itk::ImageFileWriter< ImageType >;
+    using InterpolatorType = itk::BSplineInterpolateImageFunction< ImageType, float, float >;
+    using ConstDTimage = typename ImageType::ConstPointer;
+	 /** Method for creation through the object factory. */
+    itkNewMacro( Self );//OK
+
+    /** Run-time type information (and related methods). */
+    itkTypeMacro( PointToSurfaceDistanceMetric, SingleValuedPointSetToPointSetMetric );//OK
+    /**  Initialize. */
+    virtual void Initialize() noexcept(false);
+
+    /**  Get the value for single valued optimizers. */
+    MeasureType GetValue( const TransformParametersType & parameters ) const noexcept(false);
+
+    /** Get the derivatives of the match measure. */
+    void GetDerivative( const TransformParametersType & parameters, DerivativeType & Derivative ) const noexcept(false);
+
+    /**  Get value and derivatives for multiple valued optimizers. */
+    void GetValueAndDerivative( const TransformParametersType & parameters, MeasureType & Value, DerivativeType & Derivative ) const noexcept(false);
+
+    /** Set input Segmented Image File **/
+    void SetSegImageIn( const std::string  str ) ;
+
+    /** Set input Segmented Image File **/
+    void SetDTImageIn( const std::string  str ) ;
+
+    /** Set output Distance Transform File **/
+    void SetDTImageOut(  const std::string  str );
+
+    /** Get Distance Transform Image*/
+    ConstDTimage GetDTImage() const;
+
+protected:
+
+    PointToSurfaceDistanceMetric();
+    virtual ~PointToSurfaceDistanceMetric() = default;
+    bool m_AvPointWeigh;
```

Comment:
Please initialize m_AvPointWeigh, for example as follows:

    bool m_AvPointWeigh { true };

Doing so should solve a Visual Studio 2019 IntelliSense warning:
> warning C26495: Variable '::m_AvPointWeigh' is uninitialized. Always initialize a member variable (type.6).

Reply:
It is to be initialized in the derived class. But I updated the code for the sake of the compiler, in order for it does not issue warning.

Reply:
Done

Reply:
Thanks @gokhangg 
> It is to be initialized in the derived class.

Thanks, I see now it would already be initialized in `elastix::PointToSurfaceDistanceMetric`. Still I think it's preferable to avoid compilation warnings, especially when they can be fixed easily. But I think we agree already  `:-)`

It would be nice, in my opinion, to have a GoogleTest that just tests the base class, `itk::PointToSurfaceDistanceMetric`. Then it should not depend on initialization being done by the derived class anyway.

See you on 15:00 😃